numpy/core/code_generators/numpy_api.py

```diff
@@ -370,6 +370,11 @@ def get_StealRef():
     'PyDataMem_SetHandler':                 (304,),
     'PyDataMem_GetHandler':                 (305,),
     # End 1.22 API
+    'convert_datetime_to_datetimestruct':   (307,),  # GH#21199
+    'convert_datetimestruct_to_datetime':   (308,),
+    'convert_pydatetime_to_datetimestruct': (309,),
+    'get_datetime_iso_8601_strlen':         (310,),
+    'make_iso_8601_datetime':               (311,),
```

Comment:
These names are inconsistent with the previous naming convention, camel case and Py prefix. Probably should put some work into the name , Maybe start with `PyDateTime_:)

Reply:
not my call, but ill note that as the intended consumer the python-style names are nice.  my nitpick is id like to change "datetime" -> "datetime64" in the first two convert_foo functions here.

Reply:
Maybe `NpyDateTime` is best, we do have `NpyIter` as priot art or just `NpyTime_` even.  Even just `npy_` might be OK, although that is really only used for internal functions as a pattern.
Others do use `PyArray_` like `PyArray_DatetimeToDatetimeStruct`.

`PyDateTime_` is owned used by the builtin datetime module, so we can't use it.

I am not in the mood for bike-shedding the prefix or camel vs. non-camel case, but we need a prefix in C.

Maybe there is a typical pattern to create thin wrapper functions in cython to get shorter names, since I can understand the dislike for C-style names in cython cimports.

Reply:
So...

NpyDatetime_ConvertDatetime64ToDatetimeStruct
NpyDatetime_ConvertPydatetimeToDatetimeStruct
NpyDatetime_GetDatetimeISO8601StrLen
NpyDatetime_MakeISO8601Datetime

?

Reply:
I would capitalize the T, but that would work for me.  @charris do you have any preference?

Reply:
updated+green, though some whitespace is less pretty than it used to be.  not sure if anyone cares about that

Reply:
is the whitespace a blocker here?

Reply:
The file is hashed for cversion, so it is best that it looks nice enough that folks aren't tempted to "fix" it. Apart from the ragged alignment, I wonder if the comment is needed?